### PR TITLE
feat(react-swatch-picker): Added icon, disabled state and initials

### PR DESCRIPTION
**Pull request by ValentinaKozlova**
- Added icon
- Disabled state
- Initials (feature request from partners)
- Fixed comments from #30769 

image

### CODE REVIEW

File: packages/react-components/react-swatch-picker-preview/src/components/ColorSwatch/renderColorSwatch.tsx [hunk 1]
```diff
@@ -10,5 +10,11 @@ import type { ColorSwatchState, ColorSwatchSlots } from './ColorSwatch.types';
 export const renderColorSwatch_unstable = (state: ColorSwatchState) => {
   assertSlots<ColorSwatchSlots>(state);
 
-  return <state.root />;
+  return (
+    <state.root>
+      {state.root.children}
+      {state.icon && <state.icon />}
+      {state.disabled && state.disabledIcon && <state.disabledIcon />}
```
Comment (ValentinaKozlova) [inline packages/react-components/react-swatch-picker-preview/src/components/ColorSwatch/renderColorSwatch.tsx:17]: Having together icons vs initials and disabled state will be discussed today with the designer. If we decide to change something I'll update it in the following PRs

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/be0924adb91f501d839f228cb4505c7a5509db41'>be0924adb91f501d839f228cb4505c7a5509db41</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-swatch-picker-preview/src/components/ColorSwatch/useColorSwatchStyles.styles.ts [hunk 2]
```diff
@@ -43,6 +49,12 @@ const useStyles = makeResetStyles({
     outline: 'none',
     boxShadow: `inset 0 0 0 ${tokens.strokeWidthThick} ${tokens.colorStrokeFocus2}, inset 0 0 0 ${tokens.strokeWidthThicker} ${tokens.colorStrokeFocus1}`,
   }),
+  ':disabled': {
```
Comment (layershifter) [inline packages/react-components/react-swatch-picker-preview/src/components/ColorSwatch/useColorSwatchStyles.styles.ts:52]: Let's apply this based on props instead i.e. 

```ts
makeStyles({
  disabled: {
   ':hover': {
      cursor: 'not-allowed',
      boxShadow: 'none',
    },
  }
})

//

mergeClasses(
  // ...,
  state.disabled && styles.disabled,
)
```

**Review by layershifter — APPROVED** [review]

File: packages/react-components/react-swatch-picker-preview/src/components/ColorSwatch/useColorSwatchStyles.styles.ts (hunk 2 above)
Comment (ValentinaKozlova) [reply packages/react-components/react-swatch-picker-preview/src/components/ColorSwatch/useColorSwatchStyles.styles.ts]: fixed

**Review by marcosmoura — APPROVED** [review]

**Pull request merged**